### ISSUE TITLE
Cluster API does not implement the Geo commands interface

### ISSUE THREAD

**Issue by IdanFridman** [mp911de/lettuce#154]
I am trying to figure out how to use the geolocation async cluster client of your library.

before geolocation I used it this way:

 RedisClient client = new RedisClient(REDIS_HOST, REDIS_PORT);
RedisGeoAsyncCommands<String, String>   geoAsyncCommand = client.connect().async();;

.....

I cant figure out how to do it using the RedisClusterClient. Tried this:

 RedisClusterClient clusterClient = new RedisClusterClient(new RedisURI(REDIS_HOST,REDIS_PORT,60, TimeUnit.SECONDS));

//doesnt compile
        geoAsyncCommand = clusterClient.connect().async();
//

the async client return RedisAdvancedClusterAsyncCommands which RedisGeoAsyncCommands doesnt support

Thanks,
Idan.

**Comment by mp911de** [comment]
Hi @IdanFridman 
thanks for reporting the issue. That's a bug. The `RedisClusterAsyncCommands` interface does not implement the `RedisGeoAsyncCommands` interface. You can cast the API object as workaround to `RedisGeoAsyncCommands` but it's not nice. I'll include the fix in a 3.3.2/4.0.1 release.

**Comment by IdanFridman** [comment]
Could you show an example how would cast it? btw: do I need to add into it all the cluster ip's nodes?(by constructing it with list of RedisURI's or one uri of the host would be enough? thank you.

**Comment by mp911de** [comment]
To your first question:

```java
RedisClusterClient clusterClient = new RedisClusterClient(new RedisURI(
                          REDIS_HOST,REDIS_PORT,60, TimeUnit.SECONDS));

StatefulRedisClusterConnection<String, String> clusterConnection = redisClusterClient.connect();
RedisGeoAsyncCommands<String, String> geoAsync = 
                         (RedisGeoAsyncCommands<String, String>) clusterConnection.async();

geoAsync.geoadd(...);
```

Second question:
You are required to specify at least one node. The topology is determined by the client itself but it's always safer (especially in setups where you use slaves/auto-failover) to provide more than one connection points to the clients. If one node is down and you provide multiple connection points you still are able to connect to the cluster because the nodes are used in sequence to attempt a connection.

**Comment by mp911de** [comment]
Fixed, will be released with 4.0.1.Final end of November.

**Comment by mp911de** [comment]
Released 4.0.1.Final to Maven Central
